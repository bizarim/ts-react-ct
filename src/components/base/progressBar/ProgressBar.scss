.contents-header {
    padding-bottom: 40px;

    .progress {
        width: 100%;
        height: 4px;
        border-radius: 2px;
        background-color: #f2f2f2;
        margin: 20px 0 0;

        .progress-bar {
            height: 100%;
            background-color: var(--main-bg-color);
        }
    }

    .progress-desc p {
        padding: .5rem 0;
        text-align: center;
        color: var(--main-bg-color);
        font-size: 12px;
        font-weight: 500;
        margin: 0;
        line-height: 1.5;
    }
}