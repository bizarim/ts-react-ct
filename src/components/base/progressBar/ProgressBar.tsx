import React from 'react';

export interface ProgressProps {
    value: number;
}
interface State {

}

export class ProgressBar extends React.Component<ProgressProps, State> {
    public render() {
        const { value } = this.props;
        const style = {
            width: this.makeValue(value),
        };
        return (
            <div className="contents-header">
                <div className="progress">
                    <div className="progress-bar" style={style}>

                    </div>
                </div>
                <div className="progress-desc" ><p >{this.makeValue(value)}</p></div>
            </div>
        );
    }
    private makeValue(value: number): string {
        if (value < 0) value = 0;
        else if (100 < value) value = 100;

        return `${value.toFixed(0)}%`;
    }
}

