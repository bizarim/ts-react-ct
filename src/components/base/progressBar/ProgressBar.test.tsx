import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import { ProgressProps, ProgressBar } from './ProgressBar';

configure({ adapter: new Adapter() });

const defaultProps: ProgressProps = {
    value: 0,
};

const setup = (props: Partial<ProgressProps> = {}) =>
    shallow(<ProgressBar {...{ ...defaultProps, ...props }} />);

describe('ProgressBar', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct className contents-header', () => {
        const wrapper = setup();
        expect(wrapper.hasClass('contents-header')).toBeTruthy();
        // toContainMatchingElement
    });

    it('should min max value ', () => {
        const minProps: ProgressProps = {
            value: -0,
        };
        const maxProps: ProgressProps = {
            value: 121,
        };
        const minWrapper = setup(minProps);
        const minText = minWrapper.find('p').text();
        expect(minText).toBe('0%');

        const maxWrapper = setup(maxProps);
        const maxText = maxWrapper.find('p').text();
        expect(maxText).toBe('100%');
    });


});
