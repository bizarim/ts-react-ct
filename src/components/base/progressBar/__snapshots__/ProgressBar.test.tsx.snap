// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should render 1`] = `ShallowWrapper {}`;
